### ISSUE TITLE
Capacitor Google maps marker customization

### ISSUE THREAD

**Issue by Flaviu989** [ionic-team/capacitor-plugins#1025]
### Plugin
capacitor/google-maps

### Description
I want to replace my current implementation of agm with capacitor-google-maps but I can't seem to customize it very well especially regarding the markers. 
1. When setting a marker the only option is `iconUrl` which does display an image but with it's original width and height, with no actual way of customizing the image or it's parent div since it has no class or id so I can select it
![image](https://user-images.githubusercontent.com/53371112/172218932-7b622b26-2957-4f3c-9feb-1b0888919e38.png)
2. Also I've been having no luck at removing the existing points of interest as they are rendered directly on the images of witch the map is composed of
![google maps](https://user-images.githubusercontent.com/53371112/172218903-7485b5f8-b9d1-42b0-9584-43eab9008d35.PNG)

### Platform(s)
Andoird phones and iPhones

### Preferred Solution
1. I think it would be helpful if you could also add a custom class/id when adding a marker (since I want to change it's border-radius and give it a pulse animation), and maybe a scale option:
```
this.newMap.addMarker({
  coordinate: { lat: this.userLocation.lat, lng: this.userLocation.lng },
   iconUrl: 'https://thispersondoesnotexist.com/image',
   class: 'user-image',
   id: 'user-image',
   scale: { width: 40, height: 40 }
});
```
![image](https://user-images.githubusercontent.com/53371112/172224900-823298e8-8ea1-4b89-84aa-be3aa9b921d4.png) (this is what I need)
2. As for the points of interest there should be some option to not have them pre-rendered on the map when creating the map:
```
GoogleMap.create({
    id: 'map',
    element: this.mapRef.nativeElement,
    apiKey,
    config,
    zoom: 15,
    showPointOfIntrest: false
  },
});
```

### Alternatives
Agm has `<agm-overlay>` as an alternative to `<agm-markers>` which also can also be used to create some custom markers, maybe overlays could be an alternative here as well

### Additional Context
Also there seems to be no way to remove the default UI of Map as I had to remove it like this using css:
```
.gmnoprint, .gm-control-active {
  display: none;
}
```
It would be nice if this also was an option when creating the map:
```
GoogleMap.create({
    ...
    disableDefaultUI: true
  },
});
```

**Comment by francescoklekoo** [comment]
CUSTOMIZABLE MARKER

Hi to All,
the new Capacitor Google Maps plugin (v 1.1.0) works as well but has no customizable map marker yet.

The iconUrl property works only on web platform but not on Android or iOS because the mapping between this property and the icon as image is missing (the property "icon" https://developers.google.com/android/reference/com/google/android/gms/maps/model/Marker).


To fix on iOS
For implementing it on iOS you have to modify 2 files:

1. Marker.swift (located at @capacitor/google-maps/iOS/plugin)
Add the field let image: UIImage?

In init method add these lines at the end add these lines
let _URL:URL = URL(string: self.iconUrl ?? "")!
let imageData:NSData = NSData(contentsOf: _URL)!
self.image = UIImage(data: imageData as Data)

2. Map.swift (located at @capacitor/google-maps/iOS/plugin)
In both method addMarker and addMarkers,
After the line newMarker.isDraggable = marker.draggable ?? false
Add the line newMarker.icon = marker.image

That’s all, I have already tested on iOS real device and I hope this can be useful for anyone needs use their custom markers.


To fix on Android
I have not done it yet. Can someone help me to do the same things on Android files of the plugin to support the team?


Question
Is it possible to see these fixes on the next release?


Thank you for your support

**Comment by jcesarmobile** [comment]
This issue is for allowing the customization of markers on web.
For markers not working on native there is a separate (bug) issue (this one is a feature request as markers work, but the size or styles can't be customized)
https://github.com/ionic-team/capacitor-plugins/issues/1085

**Comment by dev-MMV** [comment]
> CUSTOMIZABLE MARKER
> 
> Hi to All, the new Capacitor Google Maps plugin (v 1.1.0) works as well but has no customizable map marker yet.
> 
> The iconUrl property works only on web platform but not on Android or iOS because the mapping between this property and the icon as image is missing (the property "icon" https://developers.google.com/android/reference/com/google/android/gms/maps/model/Marker).
> 
> To fix on iOS For implementing it on iOS you have to modify 2 files:
> 
> 1. Marker.swift (located at @capacitor/google-maps/iOS/plugin)
>    Add the field let image: UIImage?
> 
> In init method add these lines at the end add these lines let _URL:URL = URL(string: self.iconUrl ?? "")! let imageData:NSData = NSData(contentsOf: _URL)! self.image = UIImage(data: imageData as Data)
> 
> 2. Map.swift (located at @capacitor/google-maps/iOS/plugin)
>    In both method addMarker and addMarkers,
>    After the line newMarker.isDraggable = marker.draggable ?? false
>    Add the line newMarker.icon = marker.image
> 
> That’s all, I have already tested on iOS real device and I hope this can be useful for anyone needs use their custom markers.
> 
> To fix on Android I have not done it yet. Can someone help me to do the same things on Android files of the plugin to support the team?
> 
> Question Is it possible to see these fixes on the next release?
> 
> Thank you for your support

for fix in android 
  https://github.com/ionic-team/capacitor-plugins/issues/1085
  
 
I confirm that it works. but it doesn't work with local urls. If the url does not exist, it generates an error that closes the application

**Comment by dev-MMV** [comment]
> 
I added a check on the validity of the url in marker.swift


       func verifyUrl(urlString: String?) -> Bool {
             guard let urlString = urlString,
                   let url = URL(string: urlString) else {
                      return false
                   }
                  return UIApplication.shared.canOpenURL(url)
       }

       if(verifyUrl(urlString: self.iconUrl ?? "")){
          let _URL:URL = URL(string: self.iconUrl ?? "")!
          let imageData:NSData = NSData(contentsOf: _URL)!
            if(imageData.length > 0){
                     self.image = UIImage(data: imageData as Data)
             } else {
                    self.image = nil
             }
       } else {
          self.image = nil
       }

**Comment by dev-MMV** [comment]
> > CUSTOMIZABLE MARKER
> > Hi to All, the new Capacitor Google Maps plugin (v 1.1.0) works as well but has no customizable map marker yet.
> > The iconUrl property works only on web platform but not on Android or iOS because the mapping between this property and the icon as image is missing (the property "icon" https://developers.google.com/android/reference/com/google/android/gms/maps/model/Marker).
> > To fix on iOS For implementing it on iOS you have to modify 2 files:
> > 
> > 1. Marker.swift (located at @capacitor/google-maps/iOS/plugin)
> >    Add the field let image: UIImage?
> > 
> > In init method add these lines at the end add these lines let _URL:URL = URL(string: self.iconUrl ?? "")! let imageData:NSData = NSData(contentsOf: _URL)! self.image = UIImage(data: imageData as Data)
> > 
> > 2. Map.swift (located at @capacitor/google-maps/iOS/plugin)
> >    In both method addMarker and addMarkers,
> >    After the line newMarker.isDraggable = marker.draggable ?? false
> >    Add the line newMarker.icon = marker.image
> > 
> > That’s all, I have already tested on iOS real device and I hope this can be useful for anyone needs use their custom markers.
> > To fix on Android I have not done it yet. Can someone help me to do the same things on Android files of the plugin to support the team?
> > Question Is it possible to see these fixes on the next release?
> > Thank you for your support
> 
> for fix in android #1085
> 
> I confirm that it works. but it doesn't work with local urls. If the url does not exist, it generates an error that closes the application

I added a check on the validity of the url in marker.swift

   func verifyUrl(urlString: String?) -> Bool {
         guard let urlString = urlString,
               let url = URL(string: urlString) else {
                  return false
               }
              return UIApplication.shared.canOpenURL(url)
   }

   if(verifyUrl(urlString: self.iconUrl ?? "")){
      let _URL:URL = URL(string: self.iconUrl ?? "")!
      let imageData:NSData = NSData(contentsOf: _URL)!
        if(imageData.length > 0){
            self.image = UIImage(data: imageData as Data)
         } else {
            self.image = nil
         }
   } else {
      self.image = nil
   }

**Comment by Flaviu989** [comment]
> This issue is for allowing the customization of markers on web. For markers not working on native there is a separate (bug) issue (this one is a feature request as markers work, but the size or styles can't be customized) #1085

Yes, it's not about the markers not working it's about not being able to customize size, style or add animations. I intend on replacing AGM with this plugin but it seems I'm stuck with AGM for the moment 😦

**Comment by jcesarmobile** [comment]
marker customization was implemented on version 4.2.0 of google-maps plugin https://github.com/ionic-team/capacitor-plugins/pull/1146
